src/main/java/org/sunbird/extension/user/impl/UserProviderRegistryImpl.java

```diff
@@ -0,0 +1,297 @@
+      Map<String, Object> resultMap =
+          (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.RESULT);
+      userProfileMap.put(
+          SunbirdExtensionConstants.REGISTRY_ID, resultMap.get(SunbirdExtensionConstants.ENTITY));
+    } else {
+      String errMsg = (String) paramsMap.get(SunbirdExtensionConstants.ERR_MSG);
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:addUser : User Registry Add Entity failed - " + errMsg,
+          LoggerEnum.ERROR.name());
+      throwUserRegistryAddEntityException();
+    }
+  }
+
+  private Map<String, Object> readUser(Map<String, Object> userIdMap) {
+
+    String accessToken = getAccessToken(userIdMap);
+    ResponseData<String> responseData = null;
+    String registryId = getRegistryId(userIdMap);
+
+    try {
+      responseData = client.readEntity(new URI(registryId), getHeader(accessToken));
+    } catch (TransformationException | URISyntaxException e) {
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:readUser : User Registry Read Entity failed ==> "
+              + e.getStackTrace(),
+          LoggerEnum.ERROR.name());
+      throwUserRegistryReadEntityException();
+    }
+
+    Map<String, Object> responseMap = getResponseMap(responseData);
+    Map<String, Object> paramsMap =
+        (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.PARAMS);
+    Map<String, Object> userMap = null;
+
+    if (SunbirdExtensionConstants.STATUS_SUCCESS.equalsIgnoreCase(
+        (String) paramsMap.get(SunbirdExtensionConstants.STATUS))) {
+      Map<String, Object> resultMap =
+          (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.RESULT);
+      String userType = getUserType(userIdMap);
+      userMap = (Map<String, Object>) resultMap.get(userType);
+      userMap =
+          TransformJsonUtil.transform(
+              userReadConfig,
+              userMap,
+              userType,
+              userEnumsConfig,
+              SunbirdExtensionConstants.OPERATION_MODE_READ);
+    } else {
+      String errMsg = (String) paramsMap.get(SunbirdExtensionConstants.ERR_MSG);
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:readUser : User Registry Read Entity failed - " + errMsg,
+          LoggerEnum.ERROR.name());
+      throwUserRegistryReadEntityException();
+    }
+
+    return userMap;
+  }
+
+  private void updateUser(Map<String, Object> userProfileMap) {
+
+    String accessToken = getAccessToken(userProfileMap);
+    Map<String, Object> userMap = getUserMapForWrite(userProfileMap);
+    ResponseData<String> responseData = null;
+
+    try {
+      responseData =
+          client.updateEntity(
+              new RequestData<>(mapper.writeValueAsString(userMap)), getHeader(accessToken));
+    } catch (JsonProcessingException | TransformationException e) {
+      ProjectLogger.log(
+          "UserProviderRegistryImpl:updateUser : User Registry Update Entity failed ==> "
```

Comment:
"UserProviderRegistryImpl:updateUser: User registry client update entity resulted in exception = "